### PR TITLE
Refactor all non-CLI functionality into a library

### PR DESCRIPTION
**Pull request by ivmaykov**
Refactor all non-CLI functionality into a library.
This is done by moving all files except main.go to a new pkg/ directory.
Directory structure and code logic were not changed.
There were a few cases in which a private function had to be made public so it could be called from main.go.

The first commit also adds utils_test.go and a unit test for CertHexFingerprint(), because this code base is sorely lacking tests. More tests should be added in the future.

### CODE REVIEW

File: pkg/pivit/utils/utils_test.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
+package utils
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestCertHexFingerprint(t *testing.T) {
```
Comment (andozw) [inline pkg/pivit/utils/utils_test.go:10]: nice

**Review by andozw — APPROVED** [review]
LGTM but get a second ack from an owner before merging

**Review by yoavamit — APPROVED** [review]
🚢

**Comment by yoavamit** [comment]
@ivmaykov I authorized the Github Actions steps for this PR, and it's now failing the nix build step.
Don't worry - it's expected.

You'd need to update the expected nix build hash as instructed [here](https://github.com/cashapp/pivit/tree/main/nix#updating-vendorhash) and add another commit :)

**Comment by ivmaykov** [comment]
@yoavamit I updated the vendor hash, can you approve kicking off the workflow when you get the chance?

**Pull request merged**